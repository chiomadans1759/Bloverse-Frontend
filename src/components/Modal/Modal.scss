@import 'sass-rem';

#bl-modal {
  color: #000000;
  .bl-modal-content {
    max-height: rem(150px);
    overflow-y: scroll;
    overflow-x: hidden;
  }
  .user-image {
    object-fit: cover;
    height: rem(40px);
    width: rem(40px);
  }
  .username {
    font-weight: 900;
    .sub-span.point {
      font-weight: 100;
    }
  }
  .sub-span {
    font-weight: 900;
  }
  .sub-span.point::before {
    content: '\25CF';
    color: #2F80ED;
    font-weight: 900;
    padding-left: rem(10px);
    padding-right: rem(10px);
  }
  .sub-text {
    color: rgba(0, 0, 0, 0.45);
  }
  .darkBackground {
    background-color: #F5F5F5;
  }
  .alt-title {
    font-size: rem(20px);
  }
  .sub-title {
    font-size: rem(16px);
  }

  .noBorderRadius {
    border-radius: 0 !important;
  }
}
