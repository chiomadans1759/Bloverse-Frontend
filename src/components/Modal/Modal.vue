<template>
  <main id="bl-modal">
    <div
      class="modal fade"
      :id="target"
      tabindex="-1"
      role="dialog"
      aria-labelledby="customModal"
      aria-hidden="true"
    >
      <div class="modal-dialog" :class="{ 'modal-dialog-centered': !modalTop }" role="document">
        <div class="modal-content" :class="{ 'noBorderRadius': noBorderRadius }">
          <div class="modal-header" :class="{ darkBackground: dark }">
            <h5 v-if="title" class="modal-title" id="customModal">{{title}}</h5>
            <div>
              <h6 v-if="altTitle">
                <strong class="alt-title">{{altTitle}}</strong>
              </h6>
              <h6 v-if="subTitle" class="mt-2 sub-title">{{subTitle}}</h6>
            </div>
            <button type="button" class="close" data-dismiss="modal" aria-label="Close">
              <span aria-hidden="true">&times;</span>
            </button>
          </div>
          <div class="modal-body">
            <slot></slot>
          </div>
          <div v-if="!hideFooter" class="modal-footer">
            <slot name="footer">
              <button type="button" class="btn btn-outline-primary" data-dismiss="modal">
                {{closeText}}
              </button>
              <button type="button" class="btn btn-primary">{{actionText}}</button>
            </slot>
          </div>
        </div>
      </div>
    </div>
  </main>
</template>

<script>
export default {
  name: "Modal",
  props: {
    title: {
      type: String,
      required: false,
      default: ""
    },
    target: {
      type: String,
      required: true,
    },
    closeText: {
      type: String,
      required: false,
      default: "Close"
    },
    actionText: {
      type: String,
      required: false,
      default: "Save changes"
    },
    actionFunction: {
      type: Function,
      required: false,
      default: function() {}
    },
    hideFooter: {
      type: Boolean,
      required: false,
      default: false
    },
    dark: {
      type: Boolean,
      required: false,
      default: false
    },
    altTitle: {
      type: String,
      required: false,
      default: ""
    },
    subTitle: {
      type: String,
      required: false,
      default: ""
    },
    modalTop: {
      type: Boolean,
      default: false
    },
    noBorderRadius: {
      type: Boolean,
      default: false,
      required: false,
    }
  }
};
</script>


<style lang="scss" scoped>
@import "./Modal.scss";
</style>

