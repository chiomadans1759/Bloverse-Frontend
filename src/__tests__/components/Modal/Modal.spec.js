import { mount } from '@vue/test-utils';
import Modal from '@/components/Modal/Modal.vue';

describe('Modal component', () => {
  it('should render properly', () => {
    const wrapper =  mount(Modal, {
      propsData: {
        target: 'target',
        modalTop: false,
      }
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render intial modal', () => {
    const wrapper =  mount(Modal, {
      propsData: {
        title: 'Modal Title',
        hideFooter: false,
        actionText: 'Save changes',
        closeText: 'Close',
        target: 'target',
        modalTop: true,
      }
    });

    expect(wrapper.props().title).toBe('Modal Title');
    expect(wrapper.props().hideFooter).toBeFalsy();
    expect(wrapper.props().actionText).toBe('Save changes');
    expect(wrapper.props().closeText).toBe('Close');
    expect(wrapper.props().target).toBe('target');
    expect(wrapper.props().modalTop).toBeTruthy();
  })
});
