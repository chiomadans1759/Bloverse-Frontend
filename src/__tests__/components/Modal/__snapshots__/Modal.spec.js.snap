// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component should render properly 1`] = `
<main
  id="bl-modal"
>
  <div
    aria-hidden="true"
    aria-labelledby="customModal"
    class="modal fade"
    id="target"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="modal-dialog modal-dialog-centered"
      role="document"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <!---->
           
          <div>
            <!---->
             
            <!---->
          </div>
           
          <button
            aria-label="Close"
            class="close"
            data-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              Ã—
            </span>
          </button>
        </div>
         
        <div
          class="modal-body"
        />
         
        <div
          class="modal-footer"
        >
          <button
            class="btn btn-outline-primary"
            data-dismiss="modal"
            type="button"
          >
            Close
          </button>
           
          <button
            class="btn btn-primary"
            type="button"
          >
            Save changes
          </button>
        </div>
      </div>
    </div>
  </div>
</main>
`;
